internal/binder/function/funcs_math.go

```diff
@@ -271,18 +272,38 @@ func registerMathFunc() {
 	builtins["log"] = builtinFunc{
 		fType: ast.FuncTypeScalar,
 		exec: func(ctx api.FunctionContext, args []interface{}) (interface{}, bool) {
-			if v, e := cast.ToFloat64(args[0], cast.CONVERT_SAMEKIND); e == nil {
-				r := math.Log10(v)
-				if math.IsNaN(r) {
-					return nil, true
-				} else {
-					return r, true
-				}
-			} else {
+			v, e := cast.ToFloat64(args[0], cast.CONVERT_SAMEKIND)
+			if e != nil {
 				return e, false
 			}
+
+			r := math.NaN()
+			switch len(args) {
+			case 1:
+				r = math.Log10(v)
+
+			case 2:
+				x, e := cast.ToFloat64(args[1], cast.CONVERT_SAMEKIND)
+				if e != nil {
+					return e, false
+				}
+				r = math.Log(x) / math.Log(v)
+			}
+
+			if !math.IsNaN(r) {
+				return r, true
+			}
+			return nil, true
```

Comment:
What's the expected return for NaN?

Reply:
Any NaN values are supposed to be returned as nil.

> If called with one argument, the function returns the decimal logarithm of X. If X is less than or equal to 0, the function returns nil; if called with two arguments, the function returns the base B logarithm of X. Returns nil if X is less than or equal to 0, or if B is less than or equal to 1.

Reply:
Is this the "standard" behavior?

Reply:
MySQL’s log function has the similar behavior: https://dev.mysql.com/doc/refman/8.0/en/mathematical-functions.html#function_log